api/v1beta1/spannerautoscaler_webhook_test.go

```diff
@@ -193,7 +185,94 @@ var _ = Describe("SpannerAutoscaler validation", func() {
 				})
 			})
 		})
-		// TODO: add more unit tests for verifying all the validation conditions
+	})
+
+	Describe("Check validation settings", func() {
+		Describe("Authentication", func() {
+			BeforeEach(func() {
+				testResource.Spec.ScaleConfig.ProcessingUnits = ScaleConfigPUs{
+					Min: 1000,
+					Max: 10000,
+				}
+			})
+
+			Context("both ImpersonateConfig and IAMKeySecret are set", func() {
+				BeforeEach(func() {
+					testResource.Spec.Authentication = Authentication{
+						ImpersonateConfig: &ImpersonateConfig{
+							TargetServiceAccount: "test-service-account",
+							Delegates:            []string{"test-delegate"},
+						},
+						IAMKeySecret: &IAMKeySecret{
+							Name: "test-service-account-secret",
+							Key:  "secret",
+						},
+					}
+				})
+
+				It("should return validation error", func() {
+					_, err := createResource(testResource)
+					Expect(err).To(HaveOccurred())
+					Expect(err.Error()).To(ContainSubstring("'impersonateConfig' and 'iamKeySecret' are mutually exclusive values"))
+				})
+			})
+		})
+
+		Describe("ScaleConfig", func() {
+			BeforeEach(func() {
+				testResource.Spec.Authentication = Authentication{
+					ImpersonateConfig: &ImpersonateConfig{
+						TargetServiceAccount: "test-service-account",
+						Delegates:            []string{"test-delegate"},
+					},
+				}
+			})
+
+			Context("max nodes are smaller than min nodes", func() {
+				BeforeEach(func() {
+					testResource.Spec.ScaleConfig.Nodes = ScaleConfigNodes{
+						Max: 1,
+						Min: 2,
+					}
+				})
```

Comment:
Nit: We can omit the setup node `BeforeEach` in `Context`s where there is just one subject node `It`.